### ISSUE TITLE
About Reg

### ISSUE THREAD

**Issue by pengcheng93** [facebookresearch/supervision-by-registration#27]
Hello！ I have a question about Reg in your paper. The Reg network directly outputs the absolute coordinates of the landmarks as the meaning of heatmaps ? And not normalized?

_Originally posted by @pengcheng93 in https://github.com/facebookresearch/supervision-by-registration/issues/25#issuecomment-431660790_

**Comment by D-X-Y** [comment]
Hi,
Not the mean of heatmaps. As shown in https://github.com/facebookresearch/supervision-by-registration/blob/master/lib/datasets/GeneralDataset.py#L161, the dataset returns a `points` variable representing the coordinates of each landmark. The Reg network directly predicts this variable `points` (after normalized into 0~1).

**Comment by pengcheng93** [comment]
> Hi,
> Not the mean of heatmaps. As shown in https://github.com/facebookresearch/supervision-by-registration/blob/master/lib/datasets/GeneralDataset.py#L161, the dataset returns a `points` variable representing the coordinates of each landmark. The Reg network directly predicts this variable `points` (after normalized into 0~1).

I understand.  Is the LK operation in Reg same as the CPM? Should the outputs of Reg be unnormalized and input to LK? Is there any change in training parameters?

**Comment by D-X-Y** [comment]
The LK operation in Reg is same as the CPM. Yes, the outputs of Reg should be unnormalized and input to LK. The parameters for LK are the same, but the optimization parameters are different.

**Comment by pengcheng93** [comment]
> The LK operation in Reg is same as the CPM. Yes, the outputs of Reg should be unnormalized and input to LK. The parameters for LK are the same, but the optimization parameters are different.

what is the optimization parameters you used? And the weight of the registration loss is 0.5  or not?

**Comment by D-X-Y** [comment]
The learning rate is 0.0001, weight decay is 0.0005, batch size is 8, LK weight is 0.5.

**Comment by pengcheng93** [comment]
> The learning rate is 0.0001, weight decay is 0.0005, batch size is 8, LK weight is 0.5.

OK, thank you very much!

**Comment by Jar7** [comment]
> The LK operation in Reg is same as the CPM. Yes, the outputs of Reg should be unnormalized and input to LK. The parameters for LK are the same, but the optimization parameters are different.

Hi. I am training SBR with a Reg Model, which is a high-precision pretrained model. In this pretrained model, the landmarks are quite stable if the bbox is stable. I add SBR to further improve the slight jitter landmarks.
I have several questions:
1. When training, I normalize the input label to 0-1. In this case, should I unnormalize the labels and output of the net to LK? Or I can use the normalized ground truth labels and outputs, just be sure they are in the same scale?

2. Would SBR improve such high-precision model, or SBR only improve model training with limited dataset?
